src/Microsoft.DotNet.Build.Tasks/PreprocessFile.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class PreprocessFile : Task
+    {
+        [Required]
+        public string SourceFile { get; set; }
+
+        [Required]
+        public string OutputFile { get; set; }
+
+        public string[] Defines { get; set; }
```

Comment:
> I would say if there is no defines present we shouldn't call the task at all.

If the file you want to pre-process has `#if` directives in it, then calling this task with no defines will strip out all the stuff guarded by the `#if`'s.  Skipping the task does something different (i.e. it keeps all the resources that should have been removed in the file).